File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode();
        dummy.next = head;
        ListNode node = head;

        int length = 1;
        while (node.next != null) {
            node = node.next;
            length++;
        }

        int curr = 0;
        node = dummy;

        while (length - n != curr) {
            node = node.next;
            curr++;
        }

        node.next = node.next.next;

        return dummy.next;
    }
}

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        heap = []
        for n in lists:
            while n:
                heapq.heappush(heap, n.val)
                n = n.next

        dummy = ListNode()
        head = dummy
        
        while heap:
            n = heapq.heappop(heap)
            ln = ListNode(n)
            head.next = ln
            head = ln

        return dummy.next